lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -313,6 +313,31 @@ class GenerateKernelExecution
     return argsCreatorFunc;
   }
 
+  /// @brief Return true if the given type corresponds to a
+  /// std-vector type according to our convention. The convention
+  /// is a ptr<struct<ptr<T>, ptr<T>, ptr<T>>>.
+  bool isStdVecArg(Type type) {
```

Comment:
Can we move this test to Factory.cpp? I know it's not building anything, but the code to build the `std::vector<T>` type is hard-coded there at the moment and having it co-located will help when we "do it right" and generate the type as specified in whatever the STL headers define.

Reply:
Can do.